presto-main/src/main/java/com/facebook/presto/type/IntervalYearMonthOperators.java

```
@@ -56,49 +58,81 @@ private IntervalYearMonthOperators()
+        }
     }
 
     @ScalarOperator(MULTIPLY)
     @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH)
     public static long multiplyByDouble(@SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long left, @SqlType(StandardTypes.DOUBLE) double right)
     {
-        return (long) (left * right);
+        long result = (long) (left * right);
+        if (result < Integer.MIN_VALUE || result > Integer.MAX_VALUE) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "Overflow multiplying interval year-month value by double: " + left + " * " + right);
+        }
+        return result;
     }
 
     @ScalarOperator(MULTIPLY)
     @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH)
-    public static long bigintMultiply(@SqlType(StandardTypes.BIGINT) long left, @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long right)
+    public static long integerMultiply(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long right)
     {
-        return left * right;
+        try {
+            return Math.multiplyExact((int) left, (int) right);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "Overflow multiplying integer by interval year-month value: " + left + " * " + right);
+        }
     }
 
     @ScalarOperator(MULTIPLY)
     @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH)
     public static long doubleMultiply(@SqlType(StandardTypes.DOUBLE) double left, @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long right)
     {
-        return (long) (left * right);
+        long result = (long) (left * right);
+        if (result < Integer.MIN_VALUE || result > Integer.MAX_VALUE) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "Overflow multiplying double by interval year-month value: " + left + " * " + right);
```

Why are we comparing the result with int_max here? The return type is long

---

As I understand, like `IntegerType`, the type `IntervalYearMonthType` in presto specifies `long.class` as its `javaType`, but the actual primitive value is int. So the functions here need to declare `long` as their parameters and return types, but the actual values should be within the range of integers. This PR handles integer overflow scenarios by directly throwing expected exceptions rather than returning unexpected values after overflow.

---

+1, thanks @hantangwangd.